torch/multiprocessing/reductions.py

```diff
@@ -595,13 +597,13 @@ def reduce_typed_storage_child(storage):
 def reduce_storage(storage):
     from . import get_sharing_strategy
 
-    if storage.is_cuda:
+    if storage.device.type == "meta":
         raise RuntimeError(
-            "Cannot pickle CUDA storage; try pickling a CUDA tensor instead"
+            "Cannot pickle meta storage; try pickling a meta tensor instead"
         )
-    elif storage.device.type == "meta":
+    elif storage.device.type != "cpu":
         raise RuntimeError(
-            "Cannot pickle meta storage; try pickling a meta tensor instead"
+            "Cannot pickle Device storage; try pickling a Device tensor instead"
```

Comment:
nit: interpolate storage.device.type here in the error message

Reply:
and then actually I guess you can just coalesce these two checks into one lol

Reply:
@ezyang Modified